### PR TITLE
2048-qt: init at 0.1.7

### PR DESCRIPTION
**Pull request by raspher**
## Description of changes
Adds 2048-qt game

**It's a fork**. As main developer was not active for a long time i've created a fork. I found it much easier to fork and merge the PR from the original repository, which fixes some bugs, than patching. (those PR's required small corrections and manual testing)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by raspher** [comment]
Result of `nixpkgs-review pr 284073` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>_2048-qt</li>
  </ul>
</details>

**Comment by raspher** [comment]
Result of `nixpkgs-review pr 284073` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>_2048-qt</li>
  </ul>
</details>

**Review by IldenH — APPROVED** [review]

**Review by IldenH — CHANGES_REQUESTED** [review]
Shouldn't it be under pkgs/games/2048-qt/default.nix instead of here?

**Comment by FliegendeWurst** [comment]
> Shouldn't it be under pkgs/games/2048-qt/default.nix instead of here?

No, new packages go in pkgs/by-name.

File: pkgs/by-name/_2/_2048-qt/package.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{ cmake
+, fetchFromGitHub
```
Comment (FliegendeWurst) [inline pkgs/by-name/_2/_2048-qt/package.nix:2]: Please format with the new formatter.

File: pkgs/by-name/_2/_2048-qt/package.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    qmake
+    wrapQtAppsHook
+  ];
+
+  preConfigure = ''
+    sed -i 's|/opt/$${TARGET}/bin|'"`echo $out`"'/bin|' deployment.pri
+  '';
+
+  postInstall = ''
+    install -D -m644 res/icons/16x16/apps/${pname}.png "$out/share/icons/hicolor/16x16/apps/${pname}.png"
```
Comment (FliegendeWurst) [inline pkgs/by-name/_2/_2048-qt/package.nix:36]: Don't use pname #277994

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Pull request closed without merging**